Feat/#12

holmir97 opened the pull request:
Concert API

holmir97 committed:
feat: hashtag model 삭제

holmir97 committed:
feat: add column interest

holmir97 committed:
feat: add column interest

holmir97 committed:
feat: add column interest

holmir97 committed:
feat: delete hastag

holmir97 committed:
feat: delete hastag

holmir97 committed:
feat: delete relation post to hashtag

holmir97 committed:
fix: concert DTO

holmir97 committed:
fix: concert DTO

holmir97 committed:
feat: concert index

holmir97 committed:
feat: concert index

holmir97 committed:
feat: concert route

holmir97 committed:
feat: concert route

holmir97 committed:
feat: concert index

holmir97 committed:
feat: concert index

holmir97 committed:
feat: concert index

holmir97 committed:
feat: concert index

holmir97 committed:
refactor: getConcertAll api

holmir97 committed:
refactor: getConcertAll api

holmir97 committed:
feat: concert detail route

holmir97 committed:
feat: concert detail route

holmir97 committed:
feat: add concert DTO

holmir97 committed:
feat: add concert DTO

holmir97 committed:
refactor: getConcertDetail api

holmir97 committed:
refactor: getConcertDetail api

holmir97 committed:
refactor: getConcertSearch api

holmir97 committed:
refactor: getConcertSearch api

holmir97 committed:
feat: concertComment route

holmir97 committed:
feat: concertComment route

holmir97 committed:
refactor: postConcertComment api

holmir97 committed:
refactor: postConcertComment api

holmir97 committed:
refactor: like, scrap api

holmir97 committed:
refactor: like, scrap api

holmir97 commented:
> Concert API

holmir97 commented:
코딩천재 리현진님 ,, 많이 참고했습니다 감사합니다..! 별다른 부분은 많이 없겠지만 우선 PR날려봅니다!

hyundang commented on src/service/concertService.ts:
```diff
@@ -1,173 +1,241 @@
+ *  @error
+ *    1. limit이 없는 경우
  */
 export const getConcertAll = async (userID, offset, limit) => {
```
파라미터 타입 지정해주면 좋을 것 같아요!! @holmir97

holmir97 replied:
파라미터 타입 빠진 부분 전부 지정하겠습니다!

hyundang commented on src/controller/concert.ts:
```diff
@@ -1,261 +1,328 @@
-      req.body.userID,
-      req.query.offset,
-      req.query.limit
+    const resData = await concertService.getConcertAll(
```
resData 타입 지정해주면 좋을 것 같아요!! @holmir97

holmir97 replied:
전부 지정 할게요!

hyundang commented on src/controller/concert.ts:
```diff
@@ -1,261 +1,328 @@
+        "요청 데이터가 부족합니다."
+      );
     }
     // 회고 전체 불러오기 성공
```
if else 형태로 [ERR_HTTP_HEADERS_SENT] 에러를 잡아야 할 것 같아요

holmir97 replied:
앗,, 빠진 부분이 있었군요 수정하겠습니다

hyundang commented on src/controller/concert.ts:
```diff
@@ -1,261 +1,328 @@
-      req.query.keyword,
-      req.query.offset,
-      req.query.limit
+    const data = await concertService.getConcertSearch(
```
중요한건 아니지만 data -> resData로 변수명을 통일하는건 어떠신가욥??

holmir97 replied:
통일 할게요~

hyundang commented on src/controller/concert.ts:
```diff
@@ -1,261 +1,328 @@
+      );
+    } else {
+      // 댓글 등록 성공
+      response.basicResponse(res, returnCode.OK, "댓글 등록 성공");
```
return code를 OK->CREATED 또는 NO_CONTENT로 바꿔도 좋을 것 같아요

holmir97 replied:
댓글, 좋아요, 스크랩 등록 성공은 모두 CREATE로 통일하였습니다!

hyundang commented on src/router/concert.ts:
```diff
@@ -0,0 +1,50 @@
+  concertController.getConcertDetailController
+);
+
+router.post(
```
router를 concert/:concertID/comment, concert/:concertID/like, concert/:concertID/scrap 으로 고쳐야 할 것 같아요
이게 더 명확한 url인 것 같아서용

holmir97 replied:
comment 한 url이 더 맞는것 같네요 수정하겠습니다!

hyundang reviewed (commented):
정말 고생많으셨습니다!! 변수명도 명확하고 코드도 깔끔하게 잘 짜신것 같아요bb😁

holmir97 committed:
fix: parameter type

holmir97 committed:
fix: parameter type

holmir97 committed:
fix: response type

holmir97 committed:
fix: response type

holmir97 committed:
fix: [ERR_HTTP_HEADERS_SENT]

holmir97 committed:
fix: [ERR_HTTP_HEADERS_SENT]

holmir97 committed:
fix: variable naming

holmir97 committed:
fix: variable naming

holmir97 committed:
fix: response message

holmir97 committed:
fix: response message

holmir97 committed:
fix: route url

holmir97 committed:
fix: route url

holmir97 committed:
fix: comment DTO

holmir97 committed:
fix: models

holmir97 committed:
fix: models

holmir97 committed:
refactor: comment structure

holmir97 committed:
refactor: comment structure

hyundang committed:
Feat/#10 (#20)

* chore: ejs 설정

* feat: challenge 등록 구현

* refactor: model import 수정

* feat: controller dto 추가

* fix: User Model에 img 필드 추가

* fix: Comment, Post model 변경

- comments 필드 추가

* feat: 댓글 등록 구현

* fix: Error [ERR_HTTP_HEADERS_SENT] 해결

* fix: Model 변경

- Badge, User, UserInfo

* feat: 좋아요 등록 구현

* feat: 스크랩 등록 구현

* style: 주석 변경

* style: 변수명 변경

* fix: model 수정

- Challenge, Concert에서  timestamp 삭제
- Challenge에서 generation 삭제
- Post에 generation 추가

* feat: commentDTO 추가

* fix: model 변경사항 반영, commentDTO 추가

* feat: 챌린지 전체 불러오기 구현

* fix: model에서 timestamp 삭제

* fix: Badge timestamp 추가

* feat: userinfo 삭제

* style: 주석, 줄바꿈

* feat: challenge api 구현

- 챌린지 검색 get
- 챌린지 상세 정보 get
- 챌린지 수정 patch

* fix: join -> require 추가

* fix: challenge patch api 수정

- 데이터 업데이트 부분 수정

* feat: challenge api 구현

- 회고 삭제
- 좋아요 취소
- 스크랩 취소

* fix: 전체 조회

- required 추가

* fix: Get challenge/:challengeID 

- isLike, isScrap 추가

* fix: post comment -> res data 삭제

* fix: 8/23일자 수정사항 반영

- Post model에 userLikes, userScraps 추가
- Comment model에 children, comment 추가
- Hashtag model 삭제
- 댓글 파싱 로직 변경
- 댓글 join 수정(required: false, order 추기)
- isLikes, isScraps 로직 수정

Commit:
Merge remote-tracking branch 'origin/feat/#12' into feat/#12
